Updating User auth from talk today; see the TODOs for considerations

jad-b opened the pull request:
Didn't make it all the way through, but I wanted to lay out the bigger ideas
- UserAuth username, user metadata, and all auth-related work
- UserPII holds all sensitive information that directly relates to a real person
- sessions.go is gone; auth tokens basically represent sessions

I think UserAuth could stand to be split into UserMeta, but I don't really care right now. What probably makes the most sense is whatever the most-common use-case for DB lookups will be (minimizing JOINs, etc.)

moogar0880 committed:
Initial commit of some auth related work

moogar0880 committed:
PII => UserSecrets

jad-b committed:
Updating User auth for talk to; see the TODOs for considerations

Didn't make it all the way through, but I wanted to lay out the bigger ideas

moogar0880 commented on users/pii.go:
```diff
@@ -0,0 +1,30 @@
+  "first_name" text NOT NULL
+  "last_name" text NOT NULL
+  "email" text NOT NULL UNIQUE
+  "enabled" bool
```
Looks like the PII struct doesn't actually have any of the fields below `email`

jad-b replied:
Yeah, probably does not. I need to finish the switch.

moogar0880 commented on users/pii.go:
```diff
@@ -0,0 +1,30 @@
+// PII (Personally Identifiable Information) has been extracted from the rest
+// of the user tables. This is in the hope of preventing anything stupid from
+// accidentally happening to it (log dumps, over-eager SELECT statements).
+type UserPII struct {
```
Do we not want a `timestamp` field for UserPII?

jad-b replied:
Depends on what the timestamp is - LastModified? CreationTime?

moogar0880 commented on users/auth.go:
```diff
@@ -1,54 +1,6 @@
 package users
```
Just to make sure I follow, `users/auth` is more of just a utility package now?

jad-b replied:
Yeah, that's how it started shaking out. Once I finished re-arranging fields between UserAuth and UserPII, I ended up with the handful of auth-related functions that you now see in `auth.go`

moogar0880 committed:
Adding new fields to ExerciseMeta struct

moogar0880 committed:
removing copy-pasta from old Python code

moogar0880 committed:
Adding documentation to exercise attributes to indicate their validity checking lists

moogar0880 committed:
Merge pull request #8 from jad-b/exercise-enhancements

[WIP] Exercise enhancements

jad-b committed:
SQL and HTTP client work on UserAuth

jad-b committed:
Authenticate function for creating Users

jad-b committed:
Ignore PSQL backups; *.template.pg

jad-b committed:
Reworking user auth. Too many changes to list

jad-b committed:
Writing /authenticate endpoint into client & server

jad-b committed:
Revamping authentication test

jad-b committed:
Add 'go install' option to Makefile; simplifications

jad-b committed:
Merge pull request #11 from jad-b/jdb/better-make

Add 'go install' option to Makefile; simplifications

jad-b committed:
Fix relative-pathing issue in compose.yml

jad-b committed:
Prefix ports with '1' in development

Trying to prevent conflicts with *other* common dev ports; 8000, 8080, 5432, etc.

jad-b committed:
Improving request feedback loop.

Incoming requests are getting logged by the RequestHandler type.
Timestamp parsing is now in line with RFC 1123
Improving host:port address validation

jad-b committed:
Add build against latest Go version in Travis

jad-b committed:
Use package names in 'make build'

jad-b committed:
Client and API server authentication handling.

Need to setup user in DB to test further

jad-b commented:
Opened this against jad-b/torque#basic-auth; going to re-open against master.
